compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2347,6 +2350,60 @@ public void visit(BLangTupleDestructure tupleDeStmt, AnalyzerData data) {
         if (type.tag != TypeTags.SEMANTIC_ERROR) {
             checkTupleVarRefEquivalency(tupleDeStmt.pos, tupleDeStmt.varRef,
                                         tupleDeStmt.expr.getBType(), tupleDeStmt.expr.pos, data);
+            if (tupleDeStmt.expr.getKind() == LIST_CONSTRUCTOR_EXPR) {
+                analyzeListConstrExprInTupleDestructure(tupleDeStmt.varRef,
+                        (BLangListConstructorExpr) tupleDeStmt.expr);
+            }
+        }
+    }
+
+    private void analyzeListConstrExprInTupleDestructure(BLangTupleVarRef tupleVarRef,
+                                                         BLangListConstructorExpr listExpr) {
+        List<BLangExpression> exprs = listExpr.exprs;
+        List<BLangExpression> tupleVars = tupleVarRef.expressions;
+
+        int nonRestTupleVarIndex = 0;
+        int noOfNonRestVars = tupleVars.size();
+
+        for (BLangExpression expr: exprs) {
+            int remainingNonRestVars = noOfNonRestVars - nonRestTupleVarIndex;
+
+            if (remainingNonRestVars == 0) {
+                return;
+            }
+
+            if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_SPREAD_OP) {
+                BLangExpression spreadOpExpr = ((BLangListConstructorSpreadOpExpr) expr).expr;
```

Comment:
Can't we simplify and reduce calls to getBType() function by  making `spreadOpType`as the variable than `spreadOpExpr`.
```suggestion
                BType spreadOpType = ((BLangListConstructorSpreadOpExpr) expr).expr.getBType();

```

Reply:
Good catch